public/functions.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * Custom functions to load before WordPress is loaded.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Detects the relative WordPress core path to use.
+ *
+ * If the 'WP_CORE_PATH_RELATIVE' constant is already defined, the function simply returns its value.
+ * Otherwise it reads the core path from the 'WP_CORE_DIR' environment variable or the 'wp-cli.yml'
+ * file in the directory located one level above.
+ *
+ * If all else fails, the function falls back to using the 'core-git/build' path.
+ *
+ * @access private
+ *
+ * @return string WordPress core path, relative to the 'public' directory.
+ */
+function _wordpressdev_detect_core_path_relative() {
+	if ( defined( 'WP_CORE_PATH_RELATIVE' ) ) {
+		return WP_CORE_PATH_RELATIVE;
+	}
+
+	// Try reading core path from 'WP_CORE_DIR' environment variable.
+	if ( false !== getenv( 'WP_CORE_DIR' ) ) {
+		$core_dir = getenv( 'WP_CORE_DIR' );
+		if ( 0 === strpos( $core_dir, dirname( __FILE__ ) ) ) {
+			return trim( str_replace( dirname( __FILE__ ), '', $core_dir ), '/' );
+		}
+	}
+
+	// Try reading core path from 'wp-cli.yml'.
+	if ( function_exists( 'yaml_parse_file' ) && file_exists( dirname( dirname( __FILE__ ) ) . '/wp-cli.yml' ) ) {
```

Comment:
Shouldn't this defer to a `wp-cli.local.yml` if it exists, since it overrides `wp-cli.yml`?

Reply:
I instead made `wp-cli.yml` a file that is gitignored anyway, similar like `.env`. It is set up in the installation process, but then can be changed to custom preferences in any way. I think for both these files this makes sense since their existence (and even existence of specific properties) is not strictly required.